### PR TITLE
Add docker image sync all tags on Jenkins

### LINKED ISSUE #3386: [Enhancement] Make ci-runner images easier to build and sync between different container registries
Hi,
After this PR:
* https://github.com/opensearch-project/opensearch-build/pull/3364

1. We need to improve how we build docker images as well as how to sync between different container registries.
2. Right now we will still build image of ci-runner and push to docker staging, then it should be easily synced to ecr staging without the need to manually sync or one by one through the docker-copy workflows.

Thanks.

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add docker image sync all tags on Jenkins

### Issues Resolved
#3386

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Waiting:
* https://github.com/opensearch-project/opensearch-build-libraries/pull/187

File: tests/jenkins/TestDockerCopy.groovy [hunk 1]
```diff
@@ -40,6 +40,7 @@ class TestDockerCopy extends BuildPipelineTest {
         super.setUp()
 
         // Variables
+        binding.setVariable('ALL_TAGS', false)
```
Comment (rishabh6788) [inline tests/jenkins/TestDockerCopy.groovy:43]: Can you add a test for `All_TAGS` as `true` as well?

**Comment by rishabh6788** [comment]
`TestPackerBuild > PackerBuildRegression FAILED` groovy test failing. @peterzhuamazon

File: tests/jenkins/TestDockerCopy.groovy (hunk 1 above)
Comment (peterzhuamazon) [reply tests/jenkins/TestDockerCopy.groovy:43]: That is already tested in the lib, basically the same outcome do we still need it here?

Comment (rishabh6788) [reply tests/jenkins/TestDockerCopy.groovy:43]: If already covered in the lib repo then it is fine. This is just to verify the integrity of jenkinsfile i believe.

Comment (peterzhuamazon) [reply tests/jenkins/TestDockerCopy.groovy:43]: Sure thanks.

Comment (gaiksaya) [reply tests/jenkins/TestDockerCopy.groovy:43]: Please add for true too! If someone increments the version we want to make sure our jenkins file acts as expected.

Comment (gaiksaya) [reply tests/jenkins/TestDockerCopy.groovy:43]: Also true is hazardous more than false so we want to make sure worst case scenario is covered as well

Comment (peterzhuamazon) [reply tests/jenkins/TestDockerCopy.groovy]: Added now.

File: jenkins/docker/docker-copy.jenkinsfile [hunk 2]
```diff
@@ -45,7 +50,7 @@ pipeline {
                 stage('Parameters Check') {
                     steps {
                         script {
-                            currentBuild.description = "${SOURCE_IMAGE_REGISTRY}/${SOURCE_IMAGE} to ${DESTINATION_IMAGE_REGISTRY}/${DESTINATION_IMAGE}"
+                            currentBuild.description = """ALL_TAGS: ${ALL_TAGS}<br>Copying: ${SOURCE_IMAGE_REGISTRY}/${SOURCE_IMAGE} to ${DESTINATION_IMAGE_REGISTRY}/${DESTINATION_IMAGE}"""
```
Comment (rishabh6788) [inline jenkins/docker/docker-copy.jenkinsfile:53]: """ is intentional or by mistake? @peterzhuamazon

File: tests/jenkins/TestDockerCopy.groovy [hunk 3]
```diff
@@ -60,8 +60,27 @@ class TestDockerCopy extends BuildPipelineTest {
         assertJobStatusSuccess()
 
         // Ensure the gcrane is executed in an external shell script exactely once
-        def copyContainerCommand = getCommands('docker').findAll {
-            shCommand -> shCommand.contains('gcrane cp opensearchstaging/opensearch:1.3.2 opensearchproject/opensearch:1.3.2; docker logout')
+        def copyContainerCommand = getCommands('crane').findAll {
+            shCommand -> shCommand.contains('set -x && crane cp opensearchstaging/ci-runner:testtag public.ecr.aws/opensearchstaging/ci-runner:testtag2')
```
Comment (rishabh6788) [inline tests/jenkins/TestDockerCopy.groovy:64]: `gcrane` to renamed to `crane`?

**Comment by rishabh6788** [comment]
General query, Can we copy images from docker to ecr and vice-versa?
In case it is not supported then we must add a check to fail the job if source and destination repos don't match, can be taken up in future if not already covered. @peterzhuamazon

File: jenkins/docker/docker-copy.jenkinsfile (hunk 2 above)
Comment (peterzhuamazon) [reply jenkins/docker/docker-copy.jenkinsfile:53]: It is intentional to parse html tags.

File: tests/jenkins/TestDockerCopy.groovy (hunk 3 above)
Comment (peterzhuamazon) [reply tests/jenkins/TestDockerCopy.groovy:64]: See explanations: https://github.com/opensearch-project/opensearch-build-libraries/pull/187#issuecomment-1541028643

**Comment by peterzhuamazon** [comment]
> General query, Can we copy images from docker to ecr and vice-versa? In case it is not supported then we must add a check to fail the job if source and destination repos don't match, can be taken up in future if not already covered. @peterzhuamazon

It is supported, that is why the test is simulating a scenario where copying from docker hub staging to ecr staging.

**Review by rishabh6788 — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
Test run: https://build.ci.opensearch.org/job/docker-copy/491/console

**Pull request merged**